### PR TITLE
158 notebooks should have access to requesters oda token

### PR DESCRIPTION
**Pull request by okolo**

### CODE REVIEW

**Review by dsavchenko — CHANGES_REQUESTED** [review]
The name of the PR is misleading for the moment. It only changes the way how the callback is passed and doesn't deal with the token. To actually use the context file to get a token, the `token.py` needs to be updated.

File: oda_api/api.py [hunk 1]
```diff
@@ -1345,17 +1345,26 @@ def from_response_json(cls, res_json, instrument, product):
...
+        context = json.load(file)
+    return context
+
 class ProgressReporter(object):
     """
     The class allows to report task progress to end user
     """
     def __init__(self):
-        self._callback = None
-        callback_file = ".oda_api_callback"  # perhaps it would be better to define this constant in a common lib
```
[the remark this answers is not in the document]
Comment (okolo) [reply oda_api/api.py:1354]: done

**Comment by okolo** [comment]
Currently test fails due to error while running Show_and_Save_Products.ipynb during request to 'https://www.astro.unige.ch/mmoda/dispatch-data' to search for thermonuclear bursts in the JEM-X light curve with error message "something went quite wrong, and we think it's not likely to recover on its own: [ RequestNotUnderstood: RequestNotUnderstood():[ InstrumentQueryBackEnd : jemx ] constructor failed: Found two non aliased identical job_id ]". The behavior is similar in master branch. I am attaching  @volodymyrss do you have any suggestions?

**Comment by volodymyrss** [comment]
> Currently test fails due to error while running Show_and_Save_Products.ipynb during request to 'https://www.astro.unige.ch/mmoda/dispatch-data' to search for thermonuclear bursts in the JEM-X light curve with error message "something went quite wrong, and we think it's not likely to recover on its own: [ RequestNotUnderstood: RequestNotUnderstood():[ InstrumentQueryBackEnd : jemx ] constructor failed: Found two non aliased identical job_id ]". The behavior is similar in master branch. @volodymyrss do you have any suggestions?

What if you just download and run the notebook, does it fail in the same way?

**Comment by okolo** [comment]
> > Currently test fails due to error while running Show_and_Save_Products.ipynb during request to 'https://www.astro.unige.ch/mmoda/dispatch-data' to search for thermonuclear bursts in the JEM-X light curve with error message "something went quite wrong, and we think it's not likely to recover on its own: [ RequestNotUnderstood: RequestNotUnderstood():[ InstrumentQueryBackEnd : jemx ] constructor failed: Found two non aliased identical job_id ]". The behavior is similar in master branch. @volodymyrss do you have any suggestions?
> 
> What if you just download and run the notebook, does it fail in the same way?

I've tried to run the notebook in jupyter. The error persists. Here is the notebook output 
[Show_and_Save_Products.html.zip](https://github.com/oda-hub/oda_api/files/15307134/Show_and_Save_Products.html.zip)

**Comment by volodymyrss** [comment]
> > > Currently test fails due to error while running Show_and_Save_Products.ipynb during request to 'https://www.astro.unige.ch/mmoda/dispatch-data' to search for thermonuclear bursts in the JEM-X light curve with error message "something went quite wrong, and we think it's not likely to recover on its own: [ RequestNotUnderstood: RequestNotUnderstood():[ InstrumentQueryBackEnd : jemx ] constructor failed: Found two non aliased identical job_id ]". The behavior is similar in master branch. @volodymyrss do you have any suggestions?
> > 
> > 
> > What if you just download and run the notebook, does it fail in the same way?
> 
> I've tried to run the notebook in jupyter. The error persists. Here is the notebook output [Show_and_Save_Products.html.zip](https://github.com/oda-hub/oda_api/files/15307134/Show_and_Save_Products.html.zip)

Looks like there was some strange behavior with this product on the live instance, must be related to a recent update.
Anyway now it's passing.

**Review by dsavchenko — APPROVED** [review]

**Review by volodymyrss — APPROVED** [review]

**Pull request merged**